src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -3105,5 +3126,28 @@ private void GenerateOneCode()
                     throw new NotImplementedException(SR.UnimplementedState);
             }
         }
+
+        private void EmitTimeoutCheck()
+        {
+            Label label = DefineLabel();
+
+            // Increment counter for each loop iteration.
+            Ldloc(_loopV);
+            Ldc(1);
+            Add();
+            Stloc(_loopV);
+
+            // Emit code to check the timeout every 2000th-iteration.
+            Ldloc(_loopV);
+            Ldc(LoopTimeoutCheckCount);
+            Rem();
```

Comment:
See comment on the non-emit version.

Reply:
This comment wasn't address.  Fine for this PR, but consider addressing it subsequently.